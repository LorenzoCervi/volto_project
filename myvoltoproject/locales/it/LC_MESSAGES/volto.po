msgid ""
msgstr ""
"Project-Id-Version: \n"
"Report-Msgid-Bugs-To: \n"
"POT-Creation-Date: \n"
"PO-Revision-Date: \n"
"Last-Translator: \n"
"Language: \n"
"Language-Team: \n"
"Content-Type: \n"
"Content-Transfer-Encoding: \n"
"Plural-Forms: \n"

#: components/Blocks/MyComp/MySelectSidebar
msgid "Age"
msgstr "Età"

#: components/Blocks/MyComp/MySelectSidebar
msgid "City"
msgstr "Città"

#: components/Blocks/MyComp/MySelectSidebar
msgid "Email"
msgstr "Email"

#: components/Blocks/MyComp/MySelectSidebar
msgid "Name"
msgstr "Nome"

#: components/Blocks/MyComp/MySelectSidebar
msgid "Notes"
msgstr "Note"

#: components/Blocks/MyComp/View
msgid "Send"
msgstr "INVIA"

#: components/Blocks/MyComp/MySelectSidebar
msgid "Sex"
msgstr "Sesso"

#: components/Blocks/MyComp/Edit
#: components/Blocks/MyComp/View
msgid "age"
msgstr "Età"

#: components/Blocks/MyComp/View
msgid "assistent"
msgstr "Parla con un Esperto"

#: components/Blocks/MyComp/Edit
#: components/Blocks/MyComp/View
msgid "city"
msgstr "Città"

#: components/Blocks/MyComp/View
msgid "errAfield"
msgstr "Errore in tutti i campi"

#: components/Blocks/MyComp/View
msgid "errAge"
msgstr "Errore contenuto Età"

#: components/Blocks/MyComp/View
msgid "errCity"
msgstr "Errore contenuto Città"

#: components/Blocks/MyComp/View
msgid "errEmail"
msgstr "Errore contenuto Email"

#: components/Blocks/MyComp/View
msgid "errName"
msgstr "Errore contenuto Nome"

#: components/Blocks/MyComp/View
msgid "errNotes"
msgstr "Errore contenuto Note"

#: components/Blocks/MyComp/View
msgid "errSex"
msgstr "Errore contenuto Sesso"

#: components/Blocks/MyComp/View
msgid "errcnt1"
msgstr "Questo input è sbagliato, leggere l'errore nel campo"

#: components/Blocks/MyComp/View
msgid "errcnt2"
msgstr "Questo campo è obbligatorio"

#: components/Blocks/MyComp/View
msgid "errcntA"
msgstr "Nessun campo richiesto è riempito,tranne Età e Motivo(default)"

#: components/Blocks/MyComp/View
msgid "female"
msgstr "Femmina"

#: components/Blocks/MyComp/View
msgid "help"
msgstr "Aiuto"

#: components/Blocks/MyComp/View
msgid "invAge"
msgstr "Età in input SBAGLIATA (range:18-100)"

#: components/Blocks/MyComp/View
msgid "invCity"
msgstr "Città in input SBAGLIATA,(deve essere maiuscola) e deve essere almeno lungo 2 lettere"

#: components/Blocks/MyComp/View
msgid "invEmail"
msgstr "Email in input SBAGLIATA (miaemail@dominio.nat)"

#: components/Blocks/MyComp/View
msgid "invName"
msgstr "Nome NON PUÒ contenere numeri o simboli e deve essere almeno lungo 2 lettere"

#: components/Blocks/MyComp/View
msgid "job"
msgstr "Richiesta lavoro"

#: components/Blocks/MyComp/View
msgid "male"
msgstr "Maschio"

#: components/Blocks/MyComp/Edit
#: components/Blocks/MyComp/View
msgid "name"
msgstr "Nome"

#: components/Blocks/MyComp/Edit
#: components/Blocks/MyComp/View
msgid "notes"
msgstr "Note"

#: components/Blocks/MyComp/View
msgid "other"
msgstr "Altro"

#: components/Blocks/MyComp/MySelectSidebar
msgid "placeholder"
msgstr "Suggerimento"

#: components/Blocks/MyComp/View
msgid "question"
msgstr "Domanda"

#: components/Blocks/MyComp/Edit
#: components/Blocks/MyComp/View
msgid "reason"
msgstr "Motivo"

#: components/Blocks/MyComp/MySelectSidebar
msgid "required"
msgstr "Campo obbligatorio"

#: components/Blocks/MyComp/Edit
#: components/Blocks/MyComp/View
msgid "sex"
msgstr "Sesso"

#: components/Blocks/MyComp/View
msgid "sexdef"
msgstr "Scegli"

#: components/Blocks/MyComp/View
msgid "succ"
msgstr "Successo!"

#: components/Blocks/MyComp/View
msgid "succSub"
msgstr "Tutti i campi sono stati inviati"
