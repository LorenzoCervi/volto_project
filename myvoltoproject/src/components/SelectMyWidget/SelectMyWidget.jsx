import React from 'react';
import {FormFieldWrapper} from '@plone/volto/components';
import loadable from '@loadable/component';

//STILE SELECT 
import {
  Option,
  DropdownIndicator,
  selectTheme,
  customSelectStyles,
} from '@plone/volto/components/manage/Widgets/SelectStyling';
//import './style.css';
import PropTypes from 'prop-types';
import Select from 'react-select';

//const Select = loadable.lib(() => import('react-select'));


const SelectMyWidget=(props)=>{
	const { 
	id,
	title,
	value,
	choices,
	required,
	onChange
	}=props;
	const inputId=`field-${id}`;
	
	return (
	
	<FormFieldWrapper {...props} className="myselect">
	  <Select
	  	id={inputId}
	  	key={props.choices}
	  	className="react-select-container"
            	classNamePrefix="react-select"
            	styles={customSelectStyles}
            	theme={selectTheme}
            	components={{ DropdownIndicator,Option }}
	  	name={id}
	  	value={value || {value:'',label:''}}
	  	options={choices}
	  	onChange={onChange}
	  />
	</FormFieldWrapper>
	);
};

SelectMyWidget.propTypes={
 id:PropTypes.string.isRequired,
 title:PropTypes.string.isRequired,
 required:PropTypes.bool,
 value:PropTypes.object,
 choices: PropTypes.arrayOf(
      PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
    ),
  onChange:PropTypes.func.isRequired,
};

SelectMyWidget.defaultProps={
};







export default SelectMyWidget;
