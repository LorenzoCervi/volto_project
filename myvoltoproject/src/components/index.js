/**
 * Add your components here.
 * @module components
 * @example
 * import Footer from './Footer/Footer';
 *
 * export {
 *   Footer,
 * };
 */
import EmailWidget from './EmailWidget/EmailWidget';
import NumberWidget from './NumberWidget/NumberWidget';

import SelectMyWidget from './SelectMyWidget/SelectMyWidget';

import MyFormView from './Blocks/MyComp/View';
import MyFormEdit from './Blocks/MyComp/Edit';

export { MyFormView, MyFormEdit, EmailWidget, NumberWidget, SelectMyWidget };
