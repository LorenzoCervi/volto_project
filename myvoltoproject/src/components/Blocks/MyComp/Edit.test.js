import React from "react";
import MyFormEdit from "./Edit.jsx";
import {render} from "@testing-library/react";
import renderer from "react-test-renderer";
import { Provider } from 'react-intl-redux';


import { TextWidget,TextareaWidget} from '@plone/volto/components';
import EmailWidget from '@package/components/EmailWidget/EmailWidget';
import NumberWidget from '@package/components/NumberWidget/NumberWidget';
import SelectMyWidget from '@package/components/SelectMyWidget/SelectMyWidget';

import configureStore from 'redux-mock-store';

const mockStore = configureStore();


const store = mockStore({
      intl: {
        locale: 'en',
        messages: {},
      },
    });

const mycomp=(
	<Provider store={store} >	
		<MyFormEdit/>
	</Provider>
);


//SNAPSHOT
test("render the entire component in a Snapshot using the view component", ()=>{
 	
	const component=renderer.create(mycomp);
	const tree=component.toJSON();
	expect(tree).toMatchSnapshot();
	});
	


//UNIT TEST
