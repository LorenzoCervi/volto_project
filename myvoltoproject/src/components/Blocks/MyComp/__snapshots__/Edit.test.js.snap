// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the entire component in a Snapshot using the view component 1`] = `
<form
  className="ui form"
  onSubmit={[Function]}
>
  <div
    className="inline field text"
    id="field-name"
  >
    <div
      className="ui grid"
    >
      <div
        className="stretched row"
      >
        <div
          className="four wide column"
        >
          <div
            className="wrapper"
          >
            <label
              htmlFor="field-name"
            >
              Name
            </label>
          </div>
        </div>
        <div
          className="eight wide column"
        >
          <div
            className="ui input"
          >
            <input
              id="field-name"
              maxLength={null}
              minLength={null}
              name="name"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              placeholder=""
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="inline field email"
    id="field-email"
  >
    <div
      className="ui grid"
    >
      <div
        className="stretched row"
      >
        <div
          className="four wide column"
        >
          <div
            className="wrapper"
          >
            <label
              htmlFor="field-email"
            >
              Email
            </label>
          </div>
        </div>
        <div
          className="eight wide column"
        >
          <div
            className="ui input"
          >
            <input
              id="field-email"
              maxLength={null}
              minLength={null}
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              type="email"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="inline field"
    id="field-age"
  >
    <div
      className="ui grid"
    >
      <div
        className="stretched row"
      >
        <div
          className="four wide column"
        >
          <div
            className="wrapper"
          >
            <label
              htmlFor="field-age"
            >
              Age
            </label>
          </div>
        </div>
        <div
          className="eight wide column"
        >
          <div
            className="ui input"
          >
            <input
              id="field-age"
              max={null}
              min={null}
              name="age"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              type="number"
              value="18"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="inline field myselect"
    id="field-sex"
  >
    <div
      className="ui grid"
    >
      <div
        className="stretched row"
      >
        <div
          className="four wide column"
        >
          <div
            className="wrapper"
          >
            <label
              htmlFor="field-sex"
            >
              Sex
            </label>
          </div>
        </div>
        <div
          className="eight wide column"
        >
          <div
            className="react-select-container css-2b097c-container"
            id="field-sex"
            onKeyDown={[Function]}
          >
            <div
              className="react-select__control css-1ard45z-control"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <div
                className="react-select__value-container react-select__value-container--has-value css-1dvrfad"
              >
                <div
                  className="react-select__single-value css-1uccc91-singleValue"
                />
                <div
                  className="css-b8ldur-Input"
                >
                  <div
                    className="react-select__input"
                    style={
                      Object {
                        "display": "inline-block",
                      }
                    }
                  >
                    <input
                      aria-autocomplete="list"
                      autoCapitalize="none"
                      autoComplete="off"
                      autoCorrect="off"
                      disabled={false}
                      id="react-select-2-input"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      spellCheck="false"
                      style={
                        Object {
                          "background": 0,
                          "border": 0,
                          "boxSizing": "content-box",
                          "color": "inherit",
                          "fontSize": "inherit",
                          "label": "input",
                          "opacity": 1,
                          "outline": 0,
                          "padding": 0,
                          "width": "1px",
                        }
                      }
                      tabIndex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style={
                        Object {
                          "height": 0,
                          "left": 0,
                          "overflow": "scroll",
                          "position": "absolute",
                          "top": 0,
                          "visibility": "hidden",
                          "whiteSpace": "pre",
                        }
                      }
                    >
                      
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="react-select__indicators css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  className="react-select__indicator-separator css-6fp5c8-indicatorSeparator"
                />
              </div>
            </div>
            <input
              name="sex"
              type="hidden"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="inline field text"
    id="field-city"
  >
    <div
      className="ui grid"
    >
      <div
        className="stretched row"
      >
        <div
          className="four wide column"
        >
          <div
            className="wrapper"
          >
            <label
              htmlFor="field-city"
            >
              City
            </label>
          </div>
        </div>
        <div
          className="eight wide column"
        >
          <div
            className="ui input"
          >
            <input
              id="field-city"
              maxLength={null}
              minLength={null}
              name="city"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              placeholder=""
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="inline required field myselect"
    id="field-reason"
  >
    <div
      className="ui grid"
    >
      <div
        className="stretched row"
      >
        <div
          className="four wide column"
        >
          <div
            className="wrapper"
          >
            <label
              htmlFor="field-reason"
            >
              Reason
            </label>
          </div>
        </div>
        <div
          className="eight wide column"
        >
          <div
            className="react-select-container css-2b097c-container"
            id="field-reason"
            onKeyDown={[Function]}
          >
            <div
              className="react-select__control css-1ard45z-control"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <div
                className="react-select__value-container react-select__value-container--has-value css-1dvrfad"
              >
                <div
                  className="react-select__single-value css-1uccc91-singleValue"
                />
                <div
                  className="css-b8ldur-Input"
                >
                  <div
                    className="react-select__input"
                    style={
                      Object {
                        "display": "inline-block",
                      }
                    }
                  >
                    <input
                      aria-autocomplete="list"
                      autoCapitalize="none"
                      autoComplete="off"
                      autoCorrect="off"
                      disabled={false}
                      id="react-select-3-input"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      spellCheck="false"
                      style={
                        Object {
                          "background": 0,
                          "border": 0,
                          "boxSizing": "content-box",
                          "color": "inherit",
                          "fontSize": "inherit",
                          "label": "input",
                          "opacity": 1,
                          "outline": 0,
                          "padding": 0,
                          "width": "1px",
                        }
                      }
                      tabIndex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style={
                        Object {
                          "height": 0,
                          "left": 0,
                          "overflow": "scroll",
                          "position": "absolute",
                          "top": 0,
                          "visibility": "hidden",
                          "whiteSpace": "pre",
                        }
                      }
                    >
                      
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="react-select__indicators css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  className="react-select__indicator-separator css-6fp5c8-indicatorSeparator"
                />
              </div>
            </div>
            <input
              name="reason"
              type="hidden"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="inline field textarea"
    id="field-notes"
  >
    <div
      className="ui grid"
    >
      <div
        className="stretched row"
      >
        <div
          className="four wide column"
        >
          <div
            className="wrapper"
          >
            <label
              htmlFor="field-notes"
            >
              Notes
            </label>
          </div>
        </div>
        <div
          className="eight wide column"
        >
          <textarea
            id="field-notes"
            name="notes"
            onChange={[Function]}
            onInput={[Function]}
            placeholder=""
            rows={3}
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</form>
`;
